.github/workflows/ci.yml

```diff
@@ -21,19 +21,44 @@ jobs:
 
   build-cache:
     needs: type-check
-    runs-on: ubuntu-20.04
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, macos-latest, windows-latest ]
+
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
         with:
           python-version: 3.7
-      - run: echo "date=$(date +'%Y-%m-%d')" >> $GITHUB_ENV
+      - name: Set up Ubuntu
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+        run: |
+          sudo apt install tesseract-ocr libtesseract-dev poppler-utils
+      - name: Set up Mac
+        if: ${{ matrix.os == 'macos-latest' }}
+        run: |
+          brew install jq tesseract
+      - name: Set up Windows
+        if: ${{ matrix.os == 'windows-latest' }}
+        run: |
+          chocolatey install jq xpdf-utils
+          chocolatey install tesseract --pre
+      - if: ${{ matrix.os != 'windows-latest' }}
+        run: echo "date=$(date +'%Y-%m-%d')" >> $GITHUB_ENV
+      - if: ${{ matrix.os == 'windows-latest' }}
+        run: echo "date=$(date +'%Y-%m-%d')" >> $env:GITHUB_ENV
       - name: Cache
         id: cache-python-env
         uses: actions/cache@v2
         with:
           path: ${{ env.pythonLocation }}
-          key: ${{ env.pythonLocation }}-${{ env.date }}-${{ hashFiles('setup.py') }}-${{ hashFiles('requirements.txt') }}-${{ hashFiles('requirements-dev.txt') }}
+          key: ${{ matrix.os }}-${{ env.pythonLocation }}-${{ env.date }}-${{ hashFiles('setup.py') }}-${{ hashFiles('requirements.txt') }}-${{ hashFiles('requirements-dev.txt') }}
+      - name: Install Pytorch on windows
+        if: ${{ matrix.os == 'windows-latest' }}
+        run: |
+          pip install torch==1.8.1+cpu -f https://download.pytorch.org/whl/lts/1.8/torch_lts.html
```

Comment:
@oryx1729 @tholor Any idea hoe to stop windows runner to install cuda version?

Reply:
Where do you see that the cuda version got installed? From the [CI logs](https://github.com/deepset-ai/haystack/runs/3619710082) it seems okay to me :thinking: 
```
Run pip install torch==1.8.1+cpu -f https://download.pytorch.org/whl/lts/1.8/torch_lts.html
  pip install torch==1.8.1+cpu -f https://download.pytorch.org/whl/lts/1.8/torch_lts.html
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    pythonLocation: C:\hostedtoolcache\windows\Python\3.7.9\x64
    date: 2021-09-16
Looking in links: https://download.pytorch.org/whl/lts/1.8/torch_lts.html
Requirement already satisfied: torch==1.8.1+cpu in c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages (1.8.1+cpu)

```

Reply:
Check this out https://github.com/deepset-ai/haystack/runs/3602760851?check_suite_focus=true
By default windows runner install cu11 version. Same issue I faced on Obsei hence used this hack there as well.

Reply:
Ok got it. So it's already resolved, but in a slightly hacky way :+1: